Src/Framework/Jobs/CurrentCorrelationFunction.py

```diff
@@ -148,15 +133,16 @@ def run_step(self, index):
             # loop over the trajectory time steps
             for i, frame in enumerate(self.configuration['frames']['value']):
                 conf = traj.configuration[frame]
-
-                try:
-                    vel = self._velocities[:, i, :]
-                except AttributeError:
-                    vel = traj.velocities[frame].array
-
-                for element, idxs in self._indexesPerElement.items():
+                if interpolationOrder == "no interpolation":
+                    velocities = traj.velocities[frame].array
+                else:
+                    velocities = numpy.empty(conf.array.shape)
+                    for axis in range(3):
+                        velocities[:,axis] = differentiate(conf.array[:,axis], order=interpolationOrder, dt=self.configuration['frames']['time_step'])

```

Comment:
Here, you are differentiating the x-coordinates of all atoms in one frame, then the y, and then the z, right? Is that acceptable? From looking at the other Jobs that use interpolation as well as the equation of 1st order interpolation, I was under the impression that interpolation has to be performed on the entire trajectory of a particle/group of particles, since it basically uses the position of particle at one time, and its position at another time. However, if your solution that only uses positional information of 1 frame is valid and correct, then it is obviously superior.